### ISSUE TITLE
Should slugs be made optional for the next generation Onion URL's?

### ISSUE THREAD

**Issue by Baccount** [micahflee/onionshare#594]
@micahflee Do you think the the slugs should be made optional for next generation onion URL's?  Here is a excerpt from [Next-Generation Hidden Services Design document](https://gitweb.torproject.org/torspec.git/tree/proposals/224-rend-spec-ng.txt#n1115)
```
2.5.1. First layer of encryption [HS-DESC-FIRST-LAYER]
   The first layer of HS descriptor encryption is designed to protect
   descriptor confidentiality against entities who don't know the blinded
   public key of the hidden service.
```

@mig5 commented on this in https://github.com/micahflee/onionshare/issues/461#issuecomment-358854192
` I am aware of the lack of need for slug (I would argue it could still be useful though in case some future vulnerability is discovered in v3 services that allowed discovery of the address - defense in depth)`

**Comment by mig5** [comment]
This will be fixed by #722 which expands 'public mode' for the Share feature (not just Receive feature), per #713 

I think it would be ok to have the slug be able to be turned on or off even for v3 (no need to *force* the slug to be off when using v3 onions - let's leave it up to the user).

**Comment by micahflee** [comment]
Slugs will indeed be made optional with v3 onion services, but here's an argument for keeping them by default.

I shared some files and ended up with this OnionShare address: `http://fzrmu6tifi6byvwhwdop7isprka6nffgw4sdpx2564wtnwkc7q5l2qad.onion/backlog-brisket`

Then I opened wireshark to sniff traffic, and tried loading that address in Chrome. Here's a screenshot of the traffic dump (github won't let me upload the .pcapng file directly to this issue, but this is just as good):

![screenshot_2018-09-17_09-42-16](https://user-images.githubusercontent.com/156128/45636930-0ea22680-ba5e-11e8-85db-8bb3e604c928.png)

Chrome makes a DNS request (plaintext, for anyone on the local network or anywhere upstream to spy on) for `fzrmu6tifi6byvwhwdop7isprka6nffgw4sdpx2564wtnwkc7q5l2qad.onion`, but since it fails to resolve a domain, it just doesn't work, and results in an error:

![screenshot_2018-09-17_09-44-10](https://user-images.githubusercontent.com/156128/45637029-4ad58700-ba5e-11e8-83bb-8843adc0f9a1.png)

However the slug, `backlog-brisket`, is nowhere in the traffic dump, since the HTTP request never actually gets sent.

I can definitely see people accidentally pasting the OnionShare address in the wrong web browser, or receiving an OnionShare address without instructions and not even realizing they need Tor Browser to begin with.

Without the slug, doing this will essentially share the address with attackers. With the slug, attackers still won't be able to access the share. For this reason, I think it makes sense to keep public mode off by default, and to keep the slug -- but to just make it optional, for genuinely public uses of OnionShare.
